reckon-core/src/main/groovy/org/ajoberstar/reckon/core/GitInventorySupplier.java

```diff
@@ -0,0 +1,262 @@
+              .flatMap(opt -> opt.isPresent() ? Stream.of(opt.get()) : Stream.empty())
+              .collect(Collectors.toSet());
+
+      Set<Version> claimedVersions =
+          taggedVersions.stream().map(TaggedVersion::getVersion).collect(Collectors.toSet());
+
+      return new VcsInventory(
+          headObjectId.getName(),
+          currentVersion,
+          baseVersion.getVersion(),
+          baseNormal.getVersion(),
+          commitsSinceBase,
+          parallelVersions,
+          claimedVersions);
+    } catch (IOException e) {
+      throw new UncheckedIOException(e);
+    }
+  }
+
+  private Set<TaggedVersion> getTaggedVersions(RevWalk walk) throws IOException {
+    Set<TaggedVersion> versions = new HashSet<>();
+
+    for (Ref ref : repo.getRefDatabase().getRefs(Constants.R_TAGS).values()) {
+      Ref tag = repo.peel(ref);
+      ObjectId objectId = tag.getPeeledObjectId();
+      RevCommit commit = walk.parseCommit(objectId);
+      tagParser
+          .apply(tag)
+          .ifPresent(
+              version -> {
+                versions.add(new TaggedVersion(version, commit));
+              });
+    }
+    return versions;
+  }
+
+  private Optional<TaggedVersion> findCurrent(RevCommit head, Stream<TaggedVersion> versions) {
+    return versions
+        .filter(version -> version.getCommit().equals(head))
+        .max(Comparator.comparing(TaggedVersion::getVersion));
+  }
+
+  private TaggedVersion findBase(RevWalk walk, RevCommit head, Stream<TaggedVersion> versions)
+      throws IOException {
+    walk.reset();
+    walk.setRevFilter(RevFilter.ALL);
+    walk.markStart(head);
+
+    Map<RevCommit, List<TaggedVersion>> versionsByCommit =
+        versions.collect(Collectors.groupingBy(TaggedVersion::getCommit));
+
+    Stream.Builder<List<TaggedVersion>> builder = Stream.builder();
+
+    for (RevCommit commit : walk) {
+      List<TaggedVersion> matches = versionsByCommit.get(commit);
+      if (matches != null) {
+        // Parents can't be "nearer". Exclude them to avoid extra walking.
+        for (RevCommit parent : commit.getParents()) {
+          walk.markUninteresting(parent);
+        }
+        builder.accept(matches);
+      }
+    }
+
+    return builder
+        .build()
+        .flatMap(List::stream)
+        .max(Comparator.comparing(TaggedVersion::getVersion))
+        .orElse(new TaggedVersion(Versions.VERSION_0, null));
+  }
+
+  private Set<TaggedVersion> findParallelCandidates(
+      RevWalk walk, RevCommit head, Set<TaggedVersion> candidates) {
+    return candidates
+        .stream()
+        .filter(candidate -> !doMergedInto(walk, head, candidate.getCommit()))
+        .filter(candidate -> !doMergedInto(walk, candidate.getCommit(), head))
+        .collect(Collectors.toSet());
+  }
+
+  private boolean doMergedInto(RevWalk walk, RevCommit base, RevCommit tip) {
+    // TODO consider something like JGit's 24 hour clock skew to eliminate candidates
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.ajoberstar.com/coding_rules#rule_key=squid%3AS1135)